lib/discordgo/message.go

```diff
@@ -115,7 +115,7 @@ type Message struct {
 	Attachments []*MessageAttachment `json:"attachments"`
 
 	// A list of components attached to the message.
-	Components []MessageComponent `json:"-"`
+	Components []unmarshalableMessageComponent `json:"components"`
```

Comment:
why not (m *Message)UnmarshalJSON the whole message? adding lowercase type in-between makes no logical sense

to get a button object from type .unmarshalableMessageComponent
`{{ index (index (getMessage nil <mID>).Components 0).MessageComponent.Components 0 | printf "%#v" }}`

 versus from .MessageComponent
` {{ index (index (g etMessage nil <mID>).Components 0).Components 0 | printf "%#v" }}`

Reply:
I wanted to do it this way, but I don't know how. I thought it was the same thing

Reply:
well, of all the code you copied from bwmarrin, UnmarshalJSON is also there. : )

Reply:
Would you please elaborate on the better solution? The issue we were encountering was that when the json was `-`, of course no data from the get request was being unmarshalled via `(s *Session) ChannelMessage`, and when the json was `components`, it failed to unmarshal because `MessageComponent` is an interface, not a struct.

Reply:
exactlly like bwmarrin has its UnmarshalJSON helpers for message and message events in events.go, and yes, exactly Components []MessageComponent `json:"-"` for type Message.... it will not panic out

Reply:
got it, I believe I have fixed this with the recent commit. `getMessage` and `.Message` now seem to return the components neatly. everything relevant should now match upstream, I agree that that is the most important thing, so thank you for helping me out with it. does it look alright to you?
thanks for your patience.